### PR TITLE
Chef server references cleanup.

### PR DESCRIPTION
**Pull request by levshvarts**
### Issues Fixed
https://github.com/edgexr/edge-cloud-platform/issues/340

### Description

Restricted cloudlet was failing to generate manifest due to leftover chef key references. Cleaned up chef references to address the failures as well as get rid of the dead code.

### CODE REVIEW

**Review by gainsley — APPROVED** [review]
Nice, thanks for fixing this!

File: pkg/platform/common/vmlayer/cloudlet.go [hunk 1]
```diff
@@ -89,9 +89,28 @@ func (v *VMPlatform) GetPlatformNodes(cloudlet *edgeproto.Cloudlet) []NodeInfo {
...
+		log.SpanLog(ctx, log.DebugLevelInfra, "using image from MEX_OS_IMAGE property", "imgFromProps", imgFromProps)
+		return imgFromProps, nil
+	}
+
+	vmBaseImage := v.VMProperties.CommonPf.PlatformConfig.CloudletVMImagePath
+	if vmBaseImage == "" {
+		return "", fmt.Errorf("Get cloudlet image failed, cloudletVMImagePath not set")
+	}
+	imageNameWithoutExt := util.RemoveExtension(filepath.Base(vmBaseImage))
+	return imageNameWithoutExt, nil
```
Comment (gainsley) [inline pkg/platform/common/vmlayer/cloudlet.go:107]: this code is duplicated on lines 115-124, I suggest you remove those lines and replace them with this new function.

**Review by gainsley — APPROVED** [review]
looks good!

File: pkg/platform/common/vmlayer/cloudlet.go (hunk 1 above)
Comment (levshvarts) [reply pkg/platform/common/vmlayer/cloudlet.go]: So I was thinking about this, but the logic becomes more complicated in `GetCloudletImageToUse`. But adding an extra return value made the code ok-ish.

File: pkg/platform/common/vmlayer/cloudlet.go [hunk 2]
```diff
@@ -89,21 +89,38 @@ func (v *VMPlatform) GetPlatformNodes(cloudlet *edgeproto.Cloudlet) []NodeInfo {
...
+func (v *VMPlatform) GetCloudletImageToUse(ctx context.Context, updateCallback edgeproto.CacheUpdateCallback) (string, error) {
+	imageNameWithoutExt, vmBaseImage, err := v.GetCloudletImageName(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// means we use image from props
+	if vmBaseImage == "" {
+		return imageNameWithoutExt, nil
+	}
```
Comment (gainsley) [inline pkg/platform/common/vmlayer/cloudlet.go:123]: You're right I was confused for a while why we need two strings. But now I think I get it. I suggest you rename `vmBaseImage` to `imagePath`. The distinction (which wasn't clear right away) is that one is just the name, and the other is the full path. Then I suggest changing this check to:
```
// if no image path, nothing to download, so just return the image name
if imagePath == "" {
    return imageNameWithoutExt, nil
}
```

**Pull request merged**